### PR TITLE
Added a new section on architecture on threads

### PR DESCRIPTION
**Pull request by oligamiq**
This pull request includes several changes to improve error handling, update dependencies, and add documentation for the `threads` module. The most important changes include adding error handling in the worker threads, updating the `package.json` dependencies, and documenting the architecture of the module.

### Error Handling Improvements:
* [`threads/src/shared_array_buffer/thread_spawn.ts`](diffhunk://#diff-504ace1ccd19027643f4ff897368d89d0f14e463c035cc470413b33a53dd6749R346): Added a try-catch block around `wasi.start` and posted error messages to the main thread if an error occurs. [[1]](diffhunk://#diff-504ace1ccd19027643f4ff897368d89d0f14e463c035cc470413b33a53dd6749R346) [[2]](diffhunk://#diff-504ace1ccd19027643f4ff897368d89d0f14e463c035cc470413b33a53dd6749R355-R362)
* [`threads/src/shared_array_buffer/worker_background/worker.ts`](diffhunk://#diff-2476739713676e949b44d21011ad5431a53228ae244f05a4eeff29f60dc2c5d9R255-R309): Added error handling logic to terminate workers and notify about errors using an allocator.

### Dependency Updates:
* [`threads/package.json`](diffhunk://#diff-7aeec3da18e2b0e1c86551d32263d711a32563d47917b3612febfab09083e690L19-R27): Updated several development dependencies to their latest versions, including `@biomejs/biome`, `@swc/cli`, `@swc/core`, `better-typescript-lib`, `typescript`, `vite`, and `vite-plugin-dts`.

### Documentation:
* [`threads/README.md`](diffhunk://#diff-fc2658b84f40130bff22114a1630b2c52c1d8c21091f8be485ab0be3728cdb52R29-R43): Added a new section on architecture, explaining the asynchronous and synchronous file access in different threads, the need for a dedicated thread for web workers, and included diagrams for better understanding.
* [`threads/RELEASE.md`](diffhunk://#diff-2eef011eefd84facff19599812053de05dbffae292e3e2f55d01dce6d01d9244R1-R5): Documented the first release and a bug fix in version 0.1.1.

### Version Update:
* [`threads/package.json`](diffhunk://#diff-7aeec3da18e2b0e1c86551d32263d711a32563d47917b3612febfab09083e690L3-R3): Updated the version from 0.1.0 to 0.1.1 to reflect the changes and bug fixes.

### Code Refactoring:
* [`threads/src/shared_array_buffer/worker_background/worker_blob.ts`](diffhunk://#diff-e591f9913c599b057a2207e27b89538b3f7852e0216a8d3a9c5514d71f24d550L3-R3): Updated the inline code for the worker blob to include error handling improvements.

### CODE REVIEW

File: threads/architecture/slide1.svg
Comment (bjorn3) [inline threads/architecture/slide1.svg:1]: What did you use to create these svg files?

Comment (oligamiq) [reply threads/architecture/slide1.svg:1]: Yes, it's powerpoint.
I reduced size to 136KiB.

Comment (oligamiq) [reply threads/architecture/slide1.svg:1]: The font was causing trouble.

**Comment by bjorn3** [comment]
Thanks!

**Comment by oligamiq** [comment]
I am not able to display it properly due to the difference in capitalization. Sorry.

**Pull request merged**